subsys/bluetooth/mesh/light_ctrl_srv.c

```diff
@@ -1777,14 +1777,13 @@ int bt_mesh_light_ctrl_srv_disable(struct bt_mesh_light_ctrl_srv *srv)
 {
 	atomic_clear_bit(&srv->flags, FLAG_CTRL_SRV_MANUALLY_ENABLED);
 
-	/* Restart resume timer even if the server has already been disabled: */
-	schedule_resume_timer(srv);
-
 	if (!is_enabled(srv)) {
 		return -EALREADY;
 	}
 
 	ctrl_disable(srv);
+	/* Restart resume timer even if the server has already been disabled: */
+	schedule_resume_timer(srv);
```

Comment:
For the comment to be correct, I think this also needs to be called inside the `if (!is_enabled(srv))` on line 1780

Reply:
You are right.
Having this call in the two places is not nice to see but it is needed.